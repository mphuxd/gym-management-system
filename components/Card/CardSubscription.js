import React from 'react';
import cx from 'classnames';
import * as Separator from '@radix-ui/react-separator';
import { CheckCircledIcon } from '@radix-ui/react-icons';
import { Button } from '@/components';

function CardSubscription({
  planName,
  planDescription,
  price,
  planLookUpKey,
  planIdValue,
  className,
  features,
  footnotes,
}) {
  const classNames = cx(className, 'flex flex-col items-center');

  return (
    <div className={classNames}>
      <div className="flex flex-col p-8 bg-white hover:bg-slate3 active:bg-slate4 outline outline-slate7 outline-[1px] focus:outline-slate8 focus:ring-2 focus:ring-blue8 rounded-2xl justify-between h-[32rem] hover:shadow-lg hover:shadow-purple6">
        <div className="description flex flex-col justify-center">
          <h2 className="text-2xl font-semibold w-fit ">{planName}</h2>
          <h3 className="mt-5">{planDescription}</h3>
          <h4 className="text-2xl font-bold mt-6">{price}</h4>
          <Separator.Root className="h-[1px] my-4" />
          <div className="flex flex-col mb-3 gap-y-1">
            {features &&
              features.map((feature) => (
                <div className="flex flex-row items-center">
                  <CheckCircledIcon width={16} height={16} className="mr-1" />
                  <span className="inline-block ">{feature}</span>
                </div>
              ))}
          </div>
        </div>
        <div className="flex flex-col">
          <form className="mt-4" action="/api/checkout_sessions" method="POST">
            {/* Stripe - Add a hidden field with the lookup_key of your Price */}
            <input type="hidden" name="lookup_key" value={planLookUpKey} />
            <input type="hidden" name="planId" value={planIdValue} />
            <button id="checkout-and-portal-button" type="submit">
              <Button as="div" intent="tertiary" size="large">
                Subscribe
              </Button>
            </button>
          </form>
          <span className="text-sm mt-4">{footnotes}</span>
        </div>
      </div>
    </div>
  );
}

export default CardSubscription;
