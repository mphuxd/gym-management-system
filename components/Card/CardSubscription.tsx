import React from 'react';
import cx from 'classnames';
import * as Separator from '@radix-ui/react-separator';
import { CheckCircledIcon } from '@radix-ui/react-icons';
import { Button } from '@/components';

export const CARD_SUBSCRIPTION_TEST_ID = 'cardSubscriptionTestId';

export interface CardSubscriptionProps {
  planName: string;
  price: string;
  planLookUpKey: string;
  planIdValue: string;
  className?: string;
  features: string[];
  footnotes: string;
}

function CardSubscription({
  planName,
  price,
  planLookUpKey,
  planIdValue,
  className,
  features,
  footnotes,
}: CardSubscriptionProps) {
  return (
    <div
      className={cx(
        className,
        'w-82 flex h-[32rem] flex-col items-center justify-between bg-layer  hover:bg-layer-hover'
      )}
      data-testid={CARD_SUBSCRIPTION_TEST_ID}
    >
      <div className="flex h-full w-full flex-col justify-between p-8">
        <div className="flex flex-col justify-center">
          <h2 className="w-fit text-2xl font-semibold">{planName}</h2>
          <h3 className="mt-6 text-2xl">{price}</h3>
          <Separator.Root className="my-6 h-px bg-border-subtle-darker" />
          <div className="mb-3 flex flex-col gap-y-2">
            {features &&
              features.map((feature) => (
                <div
                  key={feature}
                  className="flex flex-row items-center text-sm"
                >
                  <CheckCircledIcon
                    width={16}
                    height={16}
                    className="mr-1 text-brand"
                  />
                  <span className="inline-block">{`${feature}`}</span>
                </div>
              ))}
          </div>
        </div>
        <div className="flex flex-col">
          <form className="mt-4" action="/api/checkout_sessions" method="POST">
            {/* Stripe - Add a hidden field with the lookup_key of your Price */}
            <input type="hidden" name="lookup_key" value={planLookUpKey} />
            <input type="hidden" name="planId" value={planIdValue} />
            <Button
              type="submit"
              id="checkout-and-portal-button"
              intent="tertiary"
              size="large"
              length="medium"
            >
              Subscribe
            </Button>
          </form>
          <span className="mt-4 text-sm">{footnotes}</span>
        </div>
      </div>
    </div>
  );
}

export default CardSubscription;
