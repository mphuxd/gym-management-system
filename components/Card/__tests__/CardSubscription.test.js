import React from 'react';
import { render, screen } from '@testing-library/react';
import CardSubscription, {
  CARD_SUBSCRIPTION_TEST_ID,
} from '../CardSubscription';

describe('CardSimple ', () => {
  it('should contain text content', async () => {
    const planName = 'Test Plan';
    const planDescription = 'Test Plan Description';
    const price = '$1';
    const planLookUpKey = 'lookUpKey';
    const planIdValue = 'idValue';
    const className = 'test-classname';
    const features = ['Feature One', 'Feature Two', 'Feature Three'];
    const footnotes = 'Test Footnotes';
    render(
      <CardSubscription
        planName={planName}
        planDescription={planDescription}
        price={price}
        planLookUpKey={planLookUpKey}
        planIdValue={planIdValue}
        className={className}
        features={features}
        footnotes={footnotes}
      />
    );

    const card = await screen.findByTestId(CARD_SUBSCRIPTION_TEST_ID);
    expect(card).toHaveTextContent(planName);
    expect(card).toHaveTextContent(planDescription);
    expect(card).toHaveTextContent(price);
    expect(card).toHaveTextContent('Feature One');
    expect(card).toHaveTextContent('Feature Two');
    expect(card).toHaveTextContent('Feature Three');
    expect(card).toHaveTextContent(footnotes);
    expect(card).toHaveClass(className);
    expect(card).toMatchSnapshot();
  });

  it('should pass in className', async () => {
    const className = 'test-classname';
    const planName = 'Test Plan';
    const planDescription = 'Test Plan Description';
    const price = '$1';
    const planLookUpKey = 'lookUpKey';
    const planIdValue = 'idValue';
    const features = ['Feature One', 'Feature Two', 'Feature Three'];
    const footnotes = 'Test Footnotes';
    render(
      <CardSubscription
        planName={planName}
        planDescription={planDescription}
        price={price}
        planLookUpKey={planLookUpKey}
        planIdValue={planIdValue}
        className={className}
        features={features}
        footnotes={footnotes}
      />
    );

    const card = await screen.findByTestId(CARD_SUBSCRIPTION_TEST_ID);
    expect(card).toHaveClass(className);
    expect(card).toMatchSnapshot();
  });

  it('should have stripe input values', async () => {
    const planName = 'Test Plan';
    const planDescription = 'Test Plan Description';
    const price = '$1';
    const planLookUpKey = 'lookUpKey';
    const planIdValue = 'idValue';
    const className = 'test-classname';
    const features = ['Feature One', 'Feature Two', 'Feature Three'];
    const footnotes = 'Test Footnotes';
    render(
      <CardSubscription
        planName={planName}
        planDescription={planDescription}
        price={price}
        planLookUpKey={planLookUpKey}
        planIdValue={planIdValue}
        className={className}
        features={features}
        footnotes={footnotes}
      />
    );

    const card = await screen.findByTestId(CARD_SUBSCRIPTION_TEST_ID);
    expect(await screen.findByDisplayValue(planLookUpKey)).toBeInTheDocument();
    expect(await screen.findByDisplayValue(planIdValue)).toBeInTheDocument();
    expect(card).toMatchSnapshot();
  });

  it('should match snapshot', async () => {
    const planName = 'Test Plan';
    const planDescription = 'Test Plan Description';
    const price = '$1';
    const planLookUpKey = 'lookUpKey';
    const planIdValue = 'idValue';
    const className = 'test-classname';
    const features = ['Feature One', 'Feature Two', 'Feature Three'];
    const footnotes = 'Test Footnotes';
    render(
      <CardSubscription
        planName={planName}
        planDescription={planDescription}
        price={price}
        planLookUpKey={planLookUpKey}
        planIdValue={planIdValue}
        className={className}
        features={features}
        footnotes={footnotes}
      />
    );

    const card = await screen.findByTestId(CARD_SUBSCRIPTION_TEST_ID);
    expect(card).toMatchSnapshot();
  });
});
