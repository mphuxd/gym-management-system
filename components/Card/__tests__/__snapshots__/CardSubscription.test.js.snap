// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSimple  should contain text content 1`] = `
<div
  class="test-classname flex flex-col items-center"
  data-testid="cardSubscriptionTestId"
>
  <div
    class="flex flex-col p-8 bg-white hover:bg-slate3 active:bg-slate4 outline outline-slate7 outline-[1px] focus:outline-slate8 focus:ring-2 focus:ring-blue8 rounded-2xl justify-between h-[32rem] hover:shadow-lg hover:shadow-purple6"
  >
    <div
      class="description flex flex-col justify-center"
    >
      <h2
        class="text-2xl font-semibold w-fit "
      >
        Test Plan
      </h2>
      <h3
        class="mt-5"
      >
        Test Plan Description
      </h3>
      <h4
        class="text-2xl font-bold mt-6"
      >
        $1
      </h4>
      <div
        class="h-[1px] my-4"
        data-orientation="horizontal"
        role="separator"
      />
      <div
        class="flex flex-col mb-3 gap-y-1"
      >
        <div
          class="flex flex-row items-center"
        >
          <svg
            class="mr-1"
            fill="none"
            height="16"
            viewBox="0 0 15 15"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7.49991 0.877045C3.84222 0.877045 0.877075 3.84219 0.877075 7.49988C0.877075 11.1575 3.84222 14.1227 7.49991 14.1227C11.1576 14.1227 14.1227 11.1575 14.1227 7.49988C14.1227 3.84219 11.1576 0.877045 7.49991 0.877045ZM1.82708 7.49988C1.82708 4.36686 4.36689 1.82704 7.49991 1.82704C10.6329 1.82704 13.1727 4.36686 13.1727 7.49988C13.1727 10.6329 10.6329 13.1727 7.49991 13.1727C4.36689 13.1727 1.82708 10.6329 1.82708 7.49988ZM10.1589 5.53774C10.3178 5.31191 10.2636 5.00001 10.0378 4.84109C9.81194 4.68217 9.50004 4.73642 9.34112 4.96225L6.51977 8.97154L5.35681 7.78706C5.16334 7.59002 4.84677 7.58711 4.64973 7.78058C4.45268 7.97404 4.44978 8.29061 4.64325 8.48765L6.22658 10.1003C6.33054 10.2062 6.47617 10.2604 6.62407 10.2483C6.77197 10.2363 6.90686 10.1591 6.99226 10.0377L10.1589 5.53774Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="inline-block "
          >
            Feature One
          </span>
        </div>
        <div
          class="flex flex-row items-center"
        >
          <svg
            class="mr-1"
            fill="none"
            height="16"
            viewBox="0 0 15 15"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7.49991 0.877045C3.84222 0.877045 0.877075 3.84219 0.877075 7.49988C0.877075 11.1575 3.84222 14.1227 7.49991 14.1227C11.1576 14.1227 14.1227 11.1575 14.1227 7.49988C14.1227 3.84219 11.1576 0.877045 7.49991 0.877045ZM1.82708 7.49988C1.82708 4.36686 4.36689 1.82704 7.49991 1.82704C10.6329 1.82704 13.1727 4.36686 13.1727 7.49988C13.1727 10.6329 10.6329 13.1727 7.49991 13.1727C4.36689 13.1727 1.82708 10.6329 1.82708 7.49988ZM10.1589 5.53774C10.3178 5.31191 10.2636 5.00001 10.0378 4.84109C9.81194 4.68217 9.50004 4.73642 9.34112 4.96225L6.51977 8.97154L5.35681 7.78706C5.16334 7.59002 4.84677 7.58711 4.64973 7.78058C4.45268 7.97404 4.44978 8.29061 4.64325 8.48765L6.22658 10.1003C6.33054 10.2062 6.47617 10.2604 6.62407 10.2483C6.77197 10.2363 6.90686 10.1591 6.99226 10.0377L10.1589 5.53774Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="inline-block "
          >
            Feature Two
          </span>
        </div>
        <div
          class="flex flex-row items-center"
        >
          <svg
            class="mr-1"
            fill="none"
            height="16"
            viewBox="0 0 15 15"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7.49991 0.877045C3.84222 0.877045 0.877075 3.84219 0.877075 7.49988C0.877075 11.1575 3.84222 14.1227 7.49991 14.1227C11.1576 14.1227 14.1227 11.1575 14.1227 7.49988C14.1227 3.84219 11.1576 0.877045 7.49991 0.877045ZM1.82708 7.49988C1.82708 4.36686 4.36689 1.82704 7.49991 1.82704C10.6329 1.82704 13.1727 4.36686 13.1727 7.49988C13.1727 10.6329 10.6329 13.1727 7.49991 13.1727C4.36689 13.1727 1.82708 10.6329 1.82708 7.49988ZM10.1589 5.53774C10.3178 5.31191 10.2636 5.00001 10.0378 4.84109C9.81194 4.68217 9.50004 4.73642 9.34112 4.96225L6.51977 8.97154L5.35681 7.78706C5.16334 7.59002 4.84677 7.58711 4.64973 7.78058C4.45268 7.97404 4.44978 8.29061 4.64325 8.48765L6.22658 10.1003C6.33054 10.2062 6.47617 10.2604 6.62407 10.2483C6.77197 10.2363 6.90686 10.1591 6.99226 10.0377L10.1589 5.53774Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="inline-block "
          >
            Feature Three
          </span>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col"
    >
      <form
        action="/api/checkout_sessions"
        class="mt-4"
        method="POST"
      >
        <input
          name="lookup_key"
          type="hidden"
          value="lookUpKey"
        />
        <input
          name="planId"
          type="hidden"
          value="idValue"
        />
        <button
          id="checkout-and-portal-button"
          type="submit"
        >
          <div
            class="focus:outline text-blue11 outline-blue9 outline outline-[1px] hover:text-white hover:shadow-sm hover:bg-blue11 active:bg-blue11 active:text-white  px-4 py-2 w-fit rounded-md"
          >
            Subscribe
          </div>
        </button>
      </form>
      <span
        class="text-sm mt-4"
      >
        Test Footnotes
      </span>
    </div>
  </div>
</div>
`;

exports[`CardSimple  should have stripe input values 1`] = `
<div
  class="test-classname flex flex-col items-center"
  data-testid="cardSubscriptionTestId"
>
  <div
    class="flex flex-col p-8 bg-white hover:bg-slate3 active:bg-slate4 outline outline-slate7 outline-[1px] focus:outline-slate8 focus:ring-2 focus:ring-blue8 rounded-2xl justify-between h-[32rem] hover:shadow-lg hover:shadow-purple6"
  >
    <div
      class="description flex flex-col justify-center"
    >
      <h2
        class="text-2xl font-semibold w-fit "
      >
        Test Plan
      </h2>
      <h3
        class="mt-5"
      >
        Test Plan Description
      </h3>
      <h4
        class="text-2xl font-bold mt-6"
      >
        $1
      </h4>
      <div
        class="h-[1px] my-4"
        data-orientation="horizontal"
        role="separator"
      />
      <div
        class="flex flex-col mb-3 gap-y-1"
      >
        <div
          class="flex flex-row items-center"
        >
          <svg
            class="mr-1"
            fill="none"
            height="16"
            viewBox="0 0 15 15"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7.49991 0.877045C3.84222 0.877045 0.877075 3.84219 0.877075 7.49988C0.877075 11.1575 3.84222 14.1227 7.49991 14.1227C11.1576 14.1227 14.1227 11.1575 14.1227 7.49988C14.1227 3.84219 11.1576 0.877045 7.49991 0.877045ZM1.82708 7.49988C1.82708 4.36686 4.36689 1.82704 7.49991 1.82704C10.6329 1.82704 13.1727 4.36686 13.1727 7.49988C13.1727 10.6329 10.6329 13.1727 7.49991 13.1727C4.36689 13.1727 1.82708 10.6329 1.82708 7.49988ZM10.1589 5.53774C10.3178 5.31191 10.2636 5.00001 10.0378 4.84109C9.81194 4.68217 9.50004 4.73642 9.34112 4.96225L6.51977 8.97154L5.35681 7.78706C5.16334 7.59002 4.84677 7.58711 4.64973 7.78058C4.45268 7.97404 4.44978 8.29061 4.64325 8.48765L6.22658 10.1003C6.33054 10.2062 6.47617 10.2604 6.62407 10.2483C6.77197 10.2363 6.90686 10.1591 6.99226 10.0377L10.1589 5.53774Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="inline-block "
          >
            Feature One
          </span>
        </div>
        <div
          class="flex flex-row items-center"
        >
          <svg
            class="mr-1"
            fill="none"
            height="16"
            viewBox="0 0 15 15"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7.49991 0.877045C3.84222 0.877045 0.877075 3.84219 0.877075 7.49988C0.877075 11.1575 3.84222 14.1227 7.49991 14.1227C11.1576 14.1227 14.1227 11.1575 14.1227 7.49988C14.1227 3.84219 11.1576 0.877045 7.49991 0.877045ZM1.82708 7.49988C1.82708 4.36686 4.36689 1.82704 7.49991 1.82704C10.6329 1.82704 13.1727 4.36686 13.1727 7.49988C13.1727 10.6329 10.6329 13.1727 7.49991 13.1727C4.36689 13.1727 1.82708 10.6329 1.82708 7.49988ZM10.1589 5.53774C10.3178 5.31191 10.2636 5.00001 10.0378 4.84109C9.81194 4.68217 9.50004 4.73642 9.34112 4.96225L6.51977 8.97154L5.35681 7.78706C5.16334 7.59002 4.84677 7.58711 4.64973 7.78058C4.45268 7.97404 4.44978 8.29061 4.64325 8.48765L6.22658 10.1003C6.33054 10.2062 6.47617 10.2604 6.62407 10.2483C6.77197 10.2363 6.90686 10.1591 6.99226 10.0377L10.1589 5.53774Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="inline-block "
          >
            Feature Two
          </span>
        </div>
        <div
          class="flex flex-row items-center"
        >
          <svg
            class="mr-1"
            fill="none"
            height="16"
            viewBox="0 0 15 15"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7.49991 0.877045C3.84222 0.877045 0.877075 3.84219 0.877075 7.49988C0.877075 11.1575 3.84222 14.1227 7.49991 14.1227C11.1576 14.1227 14.1227 11.1575 14.1227 7.49988C14.1227 3.84219 11.1576 0.877045 7.49991 0.877045ZM1.82708 7.49988C1.82708 4.36686 4.36689 1.82704 7.49991 1.82704C10.6329 1.82704 13.1727 4.36686 13.1727 7.49988C13.1727 10.6329 10.6329 13.1727 7.49991 13.1727C4.36689 13.1727 1.82708 10.6329 1.82708 7.49988ZM10.1589 5.53774C10.3178 5.31191 10.2636 5.00001 10.0378 4.84109C9.81194 4.68217 9.50004 4.73642 9.34112 4.96225L6.51977 8.97154L5.35681 7.78706C5.16334 7.59002 4.84677 7.58711 4.64973 7.78058C4.45268 7.97404 4.44978 8.29061 4.64325 8.48765L6.22658 10.1003C6.33054 10.2062 6.47617 10.2604 6.62407 10.2483C6.77197 10.2363 6.90686 10.1591 6.99226 10.0377L10.1589 5.53774Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="inline-block "
          >
            Feature Three
          </span>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col"
    >
      <form
        action="/api/checkout_sessions"
        class="mt-4"
        method="POST"
      >
        <input
          name="lookup_key"
          type="hidden"
          value="lookUpKey"
        />
        <input
          name="planId"
          type="hidden"
          value="idValue"
        />
        <button
          id="checkout-and-portal-button"
          type="submit"
        >
          <div
            class="focus:outline text-blue11 outline-blue9 outline outline-[1px] hover:text-white hover:shadow-sm hover:bg-blue11 active:bg-blue11 active:text-white  px-4 py-2 w-fit rounded-md"
          >
            Subscribe
          </div>
        </button>
      </form>
      <span
        class="text-sm mt-4"
      >
        Test Footnotes
      </span>
    </div>
  </div>
</div>
`;

exports[`CardSimple  should match snapshot 1`] = `
<div
  class="test-classname flex flex-col items-center"
  data-testid="cardSubscriptionTestId"
>
  <div
    class="flex flex-col p-8 bg-white hover:bg-slate3 active:bg-slate4 outline outline-slate7 outline-[1px] focus:outline-slate8 focus:ring-2 focus:ring-blue8 rounded-2xl justify-between h-[32rem] hover:shadow-lg hover:shadow-purple6"
  >
    <div
      class="description flex flex-col justify-center"
    >
      <h2
        class="text-2xl font-semibold w-fit "
      >
        Test Plan
      </h2>
      <h3
        class="mt-5"
      >
        Test Plan Description
      </h3>
      <h4
        class="text-2xl font-bold mt-6"
      >
        $1
      </h4>
      <div
        class="h-[1px] my-4"
        data-orientation="horizontal"
        role="separator"
      />
      <div
        class="flex flex-col mb-3 gap-y-1"
      >
        <div
          class="flex flex-row items-center"
        >
          <svg
            class="mr-1"
            fill="none"
            height="16"
            viewBox="0 0 15 15"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7.49991 0.877045C3.84222 0.877045 0.877075 3.84219 0.877075 7.49988C0.877075 11.1575 3.84222 14.1227 7.49991 14.1227C11.1576 14.1227 14.1227 11.1575 14.1227 7.49988C14.1227 3.84219 11.1576 0.877045 7.49991 0.877045ZM1.82708 7.49988C1.82708 4.36686 4.36689 1.82704 7.49991 1.82704C10.6329 1.82704 13.1727 4.36686 13.1727 7.49988C13.1727 10.6329 10.6329 13.1727 7.49991 13.1727C4.36689 13.1727 1.82708 10.6329 1.82708 7.49988ZM10.1589 5.53774C10.3178 5.31191 10.2636 5.00001 10.0378 4.84109C9.81194 4.68217 9.50004 4.73642 9.34112 4.96225L6.51977 8.97154L5.35681 7.78706C5.16334 7.59002 4.84677 7.58711 4.64973 7.78058C4.45268 7.97404 4.44978 8.29061 4.64325 8.48765L6.22658 10.1003C6.33054 10.2062 6.47617 10.2604 6.62407 10.2483C6.77197 10.2363 6.90686 10.1591 6.99226 10.0377L10.1589 5.53774Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="inline-block "
          >
            Feature One
          </span>
        </div>
        <div
          class="flex flex-row items-center"
        >
          <svg
            class="mr-1"
            fill="none"
            height="16"
            viewBox="0 0 15 15"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7.49991 0.877045C3.84222 0.877045 0.877075 3.84219 0.877075 7.49988C0.877075 11.1575 3.84222 14.1227 7.49991 14.1227C11.1576 14.1227 14.1227 11.1575 14.1227 7.49988C14.1227 3.84219 11.1576 0.877045 7.49991 0.877045ZM1.82708 7.49988C1.82708 4.36686 4.36689 1.82704 7.49991 1.82704C10.6329 1.82704 13.1727 4.36686 13.1727 7.49988C13.1727 10.6329 10.6329 13.1727 7.49991 13.1727C4.36689 13.1727 1.82708 10.6329 1.82708 7.49988ZM10.1589 5.53774C10.3178 5.31191 10.2636 5.00001 10.0378 4.84109C9.81194 4.68217 9.50004 4.73642 9.34112 4.96225L6.51977 8.97154L5.35681 7.78706C5.16334 7.59002 4.84677 7.58711 4.64973 7.78058C4.45268 7.97404 4.44978 8.29061 4.64325 8.48765L6.22658 10.1003C6.33054 10.2062 6.47617 10.2604 6.62407 10.2483C6.77197 10.2363 6.90686 10.1591 6.99226 10.0377L10.1589 5.53774Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="inline-block "
          >
            Feature Two
          </span>
        </div>
        <div
          class="flex flex-row items-center"
        >
          <svg
            class="mr-1"
            fill="none"
            height="16"
            viewBox="0 0 15 15"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7.49991 0.877045C3.84222 0.877045 0.877075 3.84219 0.877075 7.49988C0.877075 11.1575 3.84222 14.1227 7.49991 14.1227C11.1576 14.1227 14.1227 11.1575 14.1227 7.49988C14.1227 3.84219 11.1576 0.877045 7.49991 0.877045ZM1.82708 7.49988C1.82708 4.36686 4.36689 1.82704 7.49991 1.82704C10.6329 1.82704 13.1727 4.36686 13.1727 7.49988C13.1727 10.6329 10.6329 13.1727 7.49991 13.1727C4.36689 13.1727 1.82708 10.6329 1.82708 7.49988ZM10.1589 5.53774C10.3178 5.31191 10.2636 5.00001 10.0378 4.84109C9.81194 4.68217 9.50004 4.73642 9.34112 4.96225L6.51977 8.97154L5.35681 7.78706C5.16334 7.59002 4.84677 7.58711 4.64973 7.78058C4.45268 7.97404 4.44978 8.29061 4.64325 8.48765L6.22658 10.1003C6.33054 10.2062 6.47617 10.2604 6.62407 10.2483C6.77197 10.2363 6.90686 10.1591 6.99226 10.0377L10.1589 5.53774Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="inline-block "
          >
            Feature Three
          </span>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col"
    >
      <form
        action="/api/checkout_sessions"
        class="mt-4"
        method="POST"
      >
        <input
          name="lookup_key"
          type="hidden"
          value="lookUpKey"
        />
        <input
          name="planId"
          type="hidden"
          value="idValue"
        />
        <button
          id="checkout-and-portal-button"
          type="submit"
        >
          <div
            class="focus:outline text-blue11 outline-blue9 outline outline-[1px] hover:text-white hover:shadow-sm hover:bg-blue11 active:bg-blue11 active:text-white  px-4 py-2 w-fit rounded-md"
          >
            Subscribe
          </div>
        </button>
      </form>
      <span
        class="text-sm mt-4"
      >
        Test Footnotes
      </span>
    </div>
  </div>
</div>
`;

exports[`CardSimple  should pass in className 1`] = `
<div
  class="test-classname flex flex-col items-center"
  data-testid="cardSubscriptionTestId"
>
  <div
    class="flex flex-col p-8 bg-white hover:bg-slate3 active:bg-slate4 outline outline-slate7 outline-[1px] focus:outline-slate8 focus:ring-2 focus:ring-blue8 rounded-2xl justify-between h-[32rem] hover:shadow-lg hover:shadow-purple6"
  >
    <div
      class="description flex flex-col justify-center"
    >
      <h2
        class="text-2xl font-semibold w-fit "
      >
        Test Plan
      </h2>
      <h3
        class="mt-5"
      >
        Test Plan Description
      </h3>
      <h4
        class="text-2xl font-bold mt-6"
      >
        $1
      </h4>
      <div
        class="h-[1px] my-4"
        data-orientation="horizontal"
        role="separator"
      />
      <div
        class="flex flex-col mb-3 gap-y-1"
      >
        <div
          class="flex flex-row items-center"
        >
          <svg
            class="mr-1"
            fill="none"
            height="16"
            viewBox="0 0 15 15"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7.49991 0.877045C3.84222 0.877045 0.877075 3.84219 0.877075 7.49988C0.877075 11.1575 3.84222 14.1227 7.49991 14.1227C11.1576 14.1227 14.1227 11.1575 14.1227 7.49988C14.1227 3.84219 11.1576 0.877045 7.49991 0.877045ZM1.82708 7.49988C1.82708 4.36686 4.36689 1.82704 7.49991 1.82704C10.6329 1.82704 13.1727 4.36686 13.1727 7.49988C13.1727 10.6329 10.6329 13.1727 7.49991 13.1727C4.36689 13.1727 1.82708 10.6329 1.82708 7.49988ZM10.1589 5.53774C10.3178 5.31191 10.2636 5.00001 10.0378 4.84109C9.81194 4.68217 9.50004 4.73642 9.34112 4.96225L6.51977 8.97154L5.35681 7.78706C5.16334 7.59002 4.84677 7.58711 4.64973 7.78058C4.45268 7.97404 4.44978 8.29061 4.64325 8.48765L6.22658 10.1003C6.33054 10.2062 6.47617 10.2604 6.62407 10.2483C6.77197 10.2363 6.90686 10.1591 6.99226 10.0377L10.1589 5.53774Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="inline-block "
          >
            Feature One
          </span>
        </div>
        <div
          class="flex flex-row items-center"
        >
          <svg
            class="mr-1"
            fill="none"
            height="16"
            viewBox="0 0 15 15"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7.49991 0.877045C3.84222 0.877045 0.877075 3.84219 0.877075 7.49988C0.877075 11.1575 3.84222 14.1227 7.49991 14.1227C11.1576 14.1227 14.1227 11.1575 14.1227 7.49988C14.1227 3.84219 11.1576 0.877045 7.49991 0.877045ZM1.82708 7.49988C1.82708 4.36686 4.36689 1.82704 7.49991 1.82704C10.6329 1.82704 13.1727 4.36686 13.1727 7.49988C13.1727 10.6329 10.6329 13.1727 7.49991 13.1727C4.36689 13.1727 1.82708 10.6329 1.82708 7.49988ZM10.1589 5.53774C10.3178 5.31191 10.2636 5.00001 10.0378 4.84109C9.81194 4.68217 9.50004 4.73642 9.34112 4.96225L6.51977 8.97154L5.35681 7.78706C5.16334 7.59002 4.84677 7.58711 4.64973 7.78058C4.45268 7.97404 4.44978 8.29061 4.64325 8.48765L6.22658 10.1003C6.33054 10.2062 6.47617 10.2604 6.62407 10.2483C6.77197 10.2363 6.90686 10.1591 6.99226 10.0377L10.1589 5.53774Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="inline-block "
          >
            Feature Two
          </span>
        </div>
        <div
          class="flex flex-row items-center"
        >
          <svg
            class="mr-1"
            fill="none"
            height="16"
            viewBox="0 0 15 15"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7.49991 0.877045C3.84222 0.877045 0.877075 3.84219 0.877075 7.49988C0.877075 11.1575 3.84222 14.1227 7.49991 14.1227C11.1576 14.1227 14.1227 11.1575 14.1227 7.49988C14.1227 3.84219 11.1576 0.877045 7.49991 0.877045ZM1.82708 7.49988C1.82708 4.36686 4.36689 1.82704 7.49991 1.82704C10.6329 1.82704 13.1727 4.36686 13.1727 7.49988C13.1727 10.6329 10.6329 13.1727 7.49991 13.1727C4.36689 13.1727 1.82708 10.6329 1.82708 7.49988ZM10.1589 5.53774C10.3178 5.31191 10.2636 5.00001 10.0378 4.84109C9.81194 4.68217 9.50004 4.73642 9.34112 4.96225L6.51977 8.97154L5.35681 7.78706C5.16334 7.59002 4.84677 7.58711 4.64973 7.78058C4.45268 7.97404 4.44978 8.29061 4.64325 8.48765L6.22658 10.1003C6.33054 10.2062 6.47617 10.2604 6.62407 10.2483C6.77197 10.2363 6.90686 10.1591 6.99226 10.0377L10.1589 5.53774Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="inline-block "
          >
            Feature Three
          </span>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col"
    >
      <form
        action="/api/checkout_sessions"
        class="mt-4"
        method="POST"
      >
        <input
          name="lookup_key"
          type="hidden"
          value="lookUpKey"
        />
        <input
          name="planId"
          type="hidden"
          value="idValue"
        />
        <button
          id="checkout-and-portal-button"
          type="submit"
        >
          <div
            class="focus:outline text-blue11 outline-blue9 outline outline-[1px] hover:text-white hover:shadow-sm hover:bg-blue11 active:bg-blue11 active:text-white  px-4 py-2 w-fit rounded-md"
          >
            Subscribe
          </div>
        </button>
      </form>
      <span
        class="text-sm mt-4"
      >
        Test Footnotes
      </span>
    </div>
  </div>
</div>
`;
